apps/dashboard/src/components/editor/index.js

```diff
@@ -29,15 +29,18 @@ import { hasUnpublishedChanges } from '../../util/project';
 import './style.scss';
 
 const Editor = ( { projectId } ) => {
-	const [ project, isSaved ] = useSelect( ( select ) => {
+	const [ project, isSaved, isEditDisabled ] = useSelect( ( select ) => {
 		return [
 			select( STORE_NAME ).getProject( projectId ),
 			select( STORE_NAME ).isProjectSaved(),
+			select( STORE_NAME ).isEditDisabled(),
 		];
 	} );
 
 	const projectContent = get( project, [ 'content' ], {} );
 
+	const saveDebounceTimeout = projectId ? 5000 : 500;
```

Comment:
This is a suggestion: When the project isn't created yet,  could you reduce the debounce timeout to save it as soon as possible?
One of the problems I noticed was happening because of the delayed execution of the `debounce`. It was triggering a project update with some outdated data.